xls/modules/zstd/data_generator.cc

```diff
@@ -0,0 +1,127 @@
+// Copyright 2024 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "xls/modules/zstd/data_generator.h"
+
+#include <algorithm>
+#include <array>
+#include <cstdint>
+#include <cstdio>
+#include <filesystem>  // NOLINT
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/time/time.h"
+#include "absl/types/span.h"
+#include "xls/common/file/get_runfile_path.h"
+#include "xls/common/status/status_macros.h"
+#include "xls/common/subprocess.h"
+
+namespace xls::zstd {
+
+static absl::StatusOr<std::vector<uint8_t>> ReadFileAsRawData(
+    const std::filesystem::path& path) {
+  std::ifstream file(path, std::ios::binary);
+  if (!file.is_open()) {
+    return absl::NotFoundError("Unable to open a test file");
+  }
+
+  std::vector<uint8_t> raw_data((std::istreambuf_iterator<char>(file)),
+                                (std::istreambuf_iterator<char>()));
+  return raw_data;
+}
+
+static std::string CreateNameForGeneratedFile(
+    absl::Span<std::string> args, std::string_view ext,
+    std::optional<std::string_view> prefix) {
+  std::string output;
+
+  if (prefix.has_value()) {
+    output += prefix.value();
+    output += "_";
+  }
+
+  for (auto const& x : args) {
+    output += x;
+  }
+  auto nospace_output = std::remove(output.begin(), output.end(), ' ');
```

Comment:
can you apply the fix suggested in https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-ranges.html ?

Reply:
Slight correction - can you use `absl::c_remove` rather than `std::remove`?

Reply:
hmm, there is no `absl::c_remove` (see: https://github.com/abseil/abseil-cpp/blob/master/absl/algorithm/container.h#L698)

Reply:
actually can you use `c_remove_copy` instead https://github.com/abseil/abseil-cpp/blob/9bd9a2d6dfb5365a1fea3e7e11a71556678980a8/absl/algorithm/container.h#L738 instead?

sorry for the conflicting advises, but I didn't realise there were similar issue w/ `std::ranges` usage as well.

Reply:
or maybe https://en.cppreference.com/w/cpp/container/vector/erase2 ?

Reply:
Changed to use `erase()`, also improved the filename generation for the test files (included `decodecorpus` parameters in the filename).